packages/databases-collections/src/components/create-namespace-modal.spec.tsx

```diff
@@ -0,0 +1,332 @@
+import React from 'react';
+import Sinon from 'sinon';
+import { expect } from 'chai';
+import {
+  render,
+  screen,
+  cleanup,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
+
+import { CreateNamespacePlugin } from '../..';
+import AppRegistry from 'hadron-app-registry';
+import { createNoopLogger } from '@mongodb-js/compass-logging/provider';
+import { createNoopTrack } from '@mongodb-js/compass-telemetry/provider';
+import type { ConnectionRepository } from '@mongodb-js/compass-connections/provider';
+import type { MongoDBInstance } from 'mongodb-instance-model';
+import type { WorkspacesService } from '@mongodb-js/compass-workspaces/provider';
+
+describe('CreateNamespaceModal [Component]', function () {
+  const connectionId = '12345';
+  const sandbox = Sinon.createSandbox();
+  const appRegistry = sandbox.spy(new AppRegistry());
+  const createNamespaceStub = sandbox.stub().resolves({});
+  const dataService = {
+    createCollection: createNamespaceStub,
+    configuredKMSProviders: sandbox.stub(),
+  };
+  const instance1 = {
+    on: sandbox.stub(),
+    off: sandbox.stub(),
+    removeListener: sandbox.stub(),
+    build: { version: '999.999.999' },
+    topologyDescription: { type: 'Unknown' },
+  } as unknown as MongoDBInstance;
+  const instanceModel = {
+    databases: {
+      get: function () {
+        return {
+          collections: [{ name: 'my-collection' }],
+        };
+      },
+    },
+  };
+  const connectionsManager = {
+    getDataServiceForConnection: sandbox.stub().returns(dataService),
+    on: sandbox.stub(),
+    removeListener: sandbox.stub(),
+  };
+  const instancesManager = {
+    getMongoDBInstanceForConnection: sandbox.stub().returns(instanceModel),
+    listMongoDBInstances: sandbox
+      .stub()
+      .returns(new Map([[connectionId, instance1]])),
+    on: sandbox.stub(),
+    removeListener: sandbox.stub(),
+  };
+  const favoriteQueries = {
+    getStorage: () => ({
+      loadAll: sandbox.stub().resolves([]),
+    }),
+  };
+  const pipelineStorage = {
+    loadAll: sandbox.stub().resolves([]),
+  };
+
+  const mockServices = {
+    globalAppRegistry: appRegistry,
+    logger: createNoopLogger(),
+    track: createNoopTrack(),
+    connectionsManager: connectionsManager as any,
+    connectionRepository: {
+      getConnectionInfoById: () => ({ id: connectionId }),
+    } as unknown as ConnectionRepository,
+    instancesManager: instancesManager as any,
+    queryStorage: favoriteQueries as any,
+    pipelineStorage: pipelineStorage as any,
+    workspaces: {
+      openCollectionWorkspace() {},
+    } as unknown as WorkspacesService,
+  };
```

Comment:
If you use new test helpers from @mongodb-js/comass-connections/test export you don't need to do so much test pre-setup (and it's also less clean-up for me to do 🙂)